### PR TITLE
Use CALLT

### LINKED ISSUE #416: Sync with Call Token
https://github.com/neo-project/neo-vm/pull/382

### LINKED ISSUE #418: Add methodToken
https://github.com/neo-project/neo/pull/2185

### PR DESCRIPTION
**Pull request by shargon**
Close #416 close #418

### CODE REVIEW

**Comment by shargon** [comment]
Wait for https://github.com/neo-project/neo-vm/pull/388

**Comment by shargon** [comment]
Can anyone help me with the last UT @neo-project/ngd-shanghai ? it returns two items and must be only one in this code.

```csharp
using Neo.SmartContract.Framework;
using Neo.SmartContract.Framework.Services.Neo;

namespace Neo.Compiler.MSIL.TestClasses
{
    public class Contract_CreateAndUpdate : SmartContract.Framework.SmartContract
    {
        public static int OldContract(byte[] nefFile, string manifest)
        {
            ContractManagement.Update((ByteString)nefFile, manifest);
            return 123;
        }
    }
}
```

**Comment by Tommo-L** [comment]
@ShawnYun will help fix the ut.

File: src/Neo.Compiler.MSIL/MSIL/Converter.cs [hunk 1]
```diff
@@ -35,10 +36,12 @@ public ModuleConverter(ILogger logger)
         public NeoModule outModule;
         private ILModule inModule;
         public Dictionary<ILMethod, NeoMethod> methodLink = new Dictionary<ILMethod, NeoMethod>();
+        public List<MethodToken> methodTokens = new List<MethodToken>();
 
         public NeoModule Convert(ILModule _in, ConvOption option = null)
         {
             this.inModule = _in;
+            this.methodTokens.Clear();
```
Comment (erikzhang) [inline src/Neo.Compiler.MSIL/MSIL/Converter.cs:44]: Remove this line?

Comment (shargon) [reply src/Neo.Compiler.MSIL/MSIL/Converter.cs]: Done

File: tests/Neo.SmartContract.Framework.UnitTests/Services/Neo/ContractTest.cs [hunk 2]
```diff
@@ -86,9 +88,16 @@ public void Test_Update()
...
+            var entryScript = _engine.ScriptEntry;
             var script = _engine.Build("./TestClasses/Contract_CreateAndUpdate.cs");
             var manifest = ContractManifest.FromJson(JObject.Parse(script.finalManifest));
-            var nef = new NefFile() { Script = script.finalNEFScript, Compiler = "unit-test", Version = "1.0", Tokens = System.Array.Empty<MethodToken>() };
+            var nef = new NefFile()
+            {
+                Script = script.finalNEFScript,
+                Compiler = "unit-test",
+                Version = "1.0",
+                Tokens = entryScript.nefFile.Tokens
```
Comment (Ashuaidehao) [inline tests/Neo.SmartContract.Framework.UnitTests/Services/Neo/ContractTest.cs:99]: Is it mistake for "script"?

File: tests/Neo.Compiler.MSIL.UnitTests/Utils/TestEngine.cs [hunk 3]
```diff
@@ -152,19 +161,32 @@ public int GetMethodReturnCount(string methodname)
...
 
         public EvaluationStack ExecuteTestCaseStandard(int offset, ushort rvcount, params StackItem[] args)
+        {
+            return ExecuteTestCaseStandard(offset, rvcount, ScriptEntry.nefFile, args);
+        }
+
+        public EvaluationStack ExecuteTestCaseStandard(int offset, ushort rvcount, NefFile contract, params StackItem[] args)
         {
             var context = InvocationStack.Pop();
             context = CreateContext(context.Script, (ushort)args.Length, rvcount, offset);
```
Comment (Tommo-L) [inline tests/Neo.Compiler.MSIL.UnitTests/Utils/TestEngine.cs:185]:
```suggestion
            context = CreateContext(context.Script, rvcount, offset);
```

File: tests/Neo.Compiler.MSIL.UnitTests/Utils/TestEngine.cs [hunk 4]
```diff
@@ -22,7 +22,7 @@ public class TestEngine : ApplicationEngine
         public BuildScript ScriptEntry { get; private set; }
 
         public TestEngine(TriggerType trigger = TriggerType.Application, IVerifiable verificable = null, StoreView snapshot = null, Block persistingBlock = null)
-            : base(trigger, verificable, snapshot ?? new TestSnapshot(), persistingBlock, TestGas)
+            : base(trigger, verificable, snapshot ?? new TestSnapshot(), persistingBlock, TestGas * 10)
```
Comment (Tommo-L) [inline tests/Neo.Compiler.MSIL.UnitTests/Utils/TestEngine.cs:25]:
```suggestion
            : base(trigger, verificable, snapshot ?? new TestSnapshot(), persistingBlock, TestGas)
```
?

File: tests/Neo.SmartContract.Framework.UnitTests/Services/Neo/ContractTest.cs (hunk 2 above)
Comment (Tommo-L) [inline tests/Neo.SmartContract.Framework.UnitTests/Services/Neo/ContractTest.cs:99]:
```suggestion
                Tokens = script.nefFile.Tokens
```

**Comment by Tommo-L** [comment]
@shargon  Apply these three fixes, the ut passed
image

**Comment by cloud8little** [comment]
Test with https://github.com/neo-project/neo/pull/2225, Good to merge.

**Comment by shargon** [comment]
Merge?

**Review by erikzhang — APPROVED** [review]

**Pull request merged**